ui/address/AddressSpecialTransfers.tsx

```diff
@@ -0,0 +1,153 @@
+
+import AddressCsvExportLink from './AddressCsvExportLink';
+
+const OVERLOAD_COUNT = 75;
+
+type Props = {
+  scrollRef?: React.RefObject<HTMLDivElement>;
+  overloadCount?: number;
+  resource: 'forward_transfers' | 'fee_payments';
+}
+
+const AddressSpecialTransfers = ({ scrollRef, overloadCount = OVERLOAD_COUNT, resource }: Props) => {
+  const router = useRouter();
+  const queryClient = useQueryClient();
+
+  const [ socketAlert, setSocketAlert ] = React.useState('');
+  const [ newItemsCount, setNewItemsCount ] = React.useState(0);
+
+  const isMobile = useIsMobile();
+  const currentAddress = getQueryParamString(router.query.hash);
+
+  const addressTxsQuery = useQueryWithPages({
+    resourceName: `address_${ resource }`,
+    pathParams: { hash: currentAddress },
+    scrollRef,
+    options: {
+      placeholderData: generateListStub<`address_${ typeof resource }`>(SPECIAL_TX, 50, { next_page_params: {
+        block_number: 9005713,
+        index: 5,
+        items_count: 50,
+      } }),
+    },
+  });
+
+  const handleNewSocketMessage: SocketMessage.AddressTxs['handler'] = (payload) => {
+    setSocketAlert('');
+
+    queryClient.setQueryData(
+      getResourceKey('address_txs', { pathParams: { hash: currentAddress } }),
+      (prevData: AddressTransactionsResponse | undefined) => {
+        if (!prevData) {
+          return;
+        }
+
+        const newItems: Array<Transaction> = [];
+        let newCount = 0;
+
+        payload.transactions.forEach(tx => {
+          const currIndex = prevData.items.findIndex((item) => item.hash === tx.hash);
+
+          if (currIndex > -1) {
+            prevData.items[currIndex] = tx;
+          } else {
+            if (newItems.length + prevData.items.length >= overloadCount) {
+              newCount++;
+            } else {
+              newItems.push(tx);
+            }
+          }
+        });
+        if (newCount > 0) {
+          setNewItemsCount(prev => prev + newCount);
+        }
+
+        return {
+          ...prevData,
+          items: [
+            ...newItems,
+            ...prevData.items,
+          ],
+        };
+      });
+
+  };
+
+  const handleSocketClose = React.useCallback(() => {
+    setSocketAlert('Connection is lost. Please refresh the page to load new transactions.');
+  }, []);
+
+  const handleSocketError = React.useCallback(() => {
+    setSocketAlert('An error has occurred while fetching new transactions. Please refresh the page.');
+  }, []);
+
+  const channel = useSocketChannel({
+    topic: `addresses:${ currentAddress?.toLowerCase() }`,
+    onSocketClose: handleSocketClose,
+    onSocketError: handleSocketError,
+    isDisabled: addressTxsQuery.pagination.page !== 1 || addressTxsQuery.isPlaceholderData,
+  });
+
+  useSocketMessage({
+    channel,
+    event: 'transaction',
+    handler: handleNewSocketMessage,
+  });
+
+  useSocketMessage({
+    channel,
+    event: 'pending_transaction',
+    handler: handleNewSocketMessage,
+  });
```

Comment:
Do we need to listen to the websocket here? It doesn't look like there are any events related to forward transfers or fee payments

Reply:
@yurikoinaba I thought to keep it there just in case we added web socket events for them on the future, but thinking it twice, I think for now we can remove it as it's adding complexity to the file and not bringing any value, and we could recover it from the git history if ever needed.